Automattic/wp-calypso
# Happychat: Use `lib/url` instead of duplicating it

#### Changes proposed in this Pull Request

Currently, the Happychat `Timeline` component uses a duplicate version of the Calypso URL lib's `addSchemeIfMissing` and `setUrlScheme`. This PR updates the component to use the ones from `lib/url` that are essentially the same (plus some TS support), and removes the duplicate versions.

I found this while inspecting remaining Lodash usages in inline help and Happychat - namely the `startsWith`, which we also get to remove in this PR.

#### Testing instructions

* Enable yourself as an agent in Happychat's staging HUD.
* Verify sending and receiving URLs via the chat still works the same way.
* Smoke test the timeline and verify it still works well.
